src/System.Management.Automation/utils/VTUtils.cs

```diff
@@ -1,83 +1,115 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-using System.Collections.Generic;
+using System;
 
 namespace System.Management.Automation
 {
     /// <summary>
     /// Class to help with VT escape sequences.
     /// </summary>
-    public sealed class VTUtility
+    public static class VTUtility
     {
+        private static readonly string[] BackgroundColorMap =
+            {
+                "\x1b[40m", // Black
+                "\x1b[44m", // DarkBlue
+                "\x1b[42m", // DarkGreen
+                "\x1b[46m", // DarkCyan
+                "\x1b[41m", // DarkRed
+                "\x1b[45m", // DarkMagenta
+                "\x1b[43m", // DarkYellow
+                "\x1b[47m", // Gray
+                "\x1b[100m", // DarkGray
+                "\x1b[104m", // Blue
+                "\x1b[102m", // Green
+                "\x1b[106m", // Cyan
+                "\x1b[101m", // Red
+                "\x1b[105m", // Magenta
+                "\x1b[103m", // Yellow
+                "\x1b[107m", // White
+            };
+
+        private static readonly string[] ForegroundColorMap =
+            {
+                "\x1b[30m", // Black
+                "\x1b[34m", // DarkBlue
+                "\x1b[32m", // DarkGreen
+                "\x1b[36m", // DarkCyan
+                "\x1b[31m", // DarkRed
+                "\x1b[35m", // DarkMagenta
+                "\x1b[33m", // DarkYellow
+                "\x1b[37m", // Gray
+                "\x1b[90m", // DarkGray
+                "\x1b[94m", // Blue
+                "\x1b[92m", // Green
+                "\x1b[96m", // Cyan
+                "\x1b[91m", // Red
+                "\x1b[95m", // Magenta
+                "\x1b[93m", // Yellow
+                "\x1b[97m", // White
+            };
+
         /// <summary>
-        /// Available VT escape codes other than colors.
+        /// Return the VT escape sequence for a ConsoleColor.
         /// </summary>
-        public enum VT
+        /// <param name="color">The <see cref="ConsoleColor"/> to be mapped from.</param>
+        /// <param name="isBackground">Whether or not it's a background color.</param>
+        /// <returns>The VT escape sequence representing the color. Or, an empty string if the passed-in color is invalid.</returns>
+        internal static string MapColorToEscapeSequence(ConsoleColor color, bool isBackground)
         {
-            /// <summary>Reset the text style.</summary>
-            Reset,
+            int index = (int)color;
+            if (index < 0 || index >= ForegroundColorMap.Length)
+            {
+                // Return empty string if the passed-in console color is out of bound.
+                return string.Empty;
```

Comment:
Maybe return a good default?
https://github.com/dotnet/runtime/blob/215b39abf947da7a40b0cb137eab4bceb24ad3e3/src/libraries/Microsoft.Extensions.Logging.Console/src/AnsiParser.cs#L130-L131

And perhaps we could be interesting in the AnsiParser.

Reply:
I'm more inclined to throw exception in case that invalid parameters are provided. Now it returns empty string only to be consistent with the existing `GetEscapeSequence` method, but given that we want to move this to `PSStyle`, I think we should throw exception in case passed-in console color is out of bound.

Reply:
Updated the code to throw exception in case of invalid `ConsoleColor` value.